### PR TITLE
v7Browser Graph display

### PR DESCRIPTION
**Pull request by Falcort**
This PR add the possibility to read a root file graph and display it into the javascript browser

### CODE REVIEW

**Comment by bellenot** [comment]
@phsft-bot build with flags -DCMAKE_CXX_STANDARD=14 -Droot7=ON

File: gui/browserv7/src/RBrowser.cxx [hunk 1]
```diff
@@ -326,14 +326,55 @@ std::string ROOT::Experimental::RBrowser::ProcessBrowserRequest(const std::strin
...
    if (request->sort == "DBLCLK") {
-      res = "FREAD:";
-      if (request->path.size() > 5 && request->path.compare(request->path.size() - 5, 5, ".root") == 0) {
-
-      } else {
-         std::ifstream t(request->path);
-         std::string str((std::istreambuf_iterator<char>(t)),
-                          std::istreambuf_iterator<char>());
-         res.append(str.c_str());
+       if(request->path.find(".root") != std::string::npos) {
```
Comment (bellenot) [inline gui/browserv7/src/RBrowser.cxx]: add a space between `if` and `(`

File: gui/browserv7/src/RBrowser.cxx [hunk 2]
```diff
@@ -326,14 +326,55 @@ std::string ROOT::Experimental::RBrowser::ProcessBrowserRequest(const std::strin
...
+
+           //Iterate over the split
+           // The goal is to have two parts
+           // The first one is the relative path of the root file to open it (rootFilePath)
+           // And the second if the name of the namecysle (rootFileName)
+            for(std::vector<int>::size_type i=0; i!=split.size(); i++) {
+                // If the current split contain .root
+                if(split[i].find(".root") != std::string::npos) {
+                    rootFilePath += split[i]; // Add the file to the path
+                        if(split[i+1].find("ntuple") != std::string::npos) {
```
Comment (bellenot) [inline gui/browserv7/src/RBrowser.cxx]: Same here: spaces between `if` and `(`

File: ui5/browser/view/Browser.view.xml [hunk 3]
```diff
@@ -61,9 +62,9 @@
                   </content>
                 </TabContainerItem>
                 <TabContainerItem name="ROOT Canvas" icon="sap-icon://column-chart-dual-axis" additionalText="" >
-                  <content>
-                        
-                  </content>
+                    <content>
+                        <core:HTML id="aRootCanvas1" content="&lt;div style=&quot;height:100%&quot;&gt;{/rootCanvas}&lt;/div&gt;" />
+                    </content>
```
Comment (bellenot) [inline ui5/browser/view/Browser.view.xml]: Indentation

File: ui5/browser/controller/Browser.controller.js [hunk 4]
```diff
@@ -298,9 +302,39 @@ sap.ui.define(['sap/ui/core/Component',
...
+
+             let oTabElement = oCanvas.getParent(); // Get the tab from the drawing place
+             let rootFileDisplayName = rootFileArray[i] + "/" + rootFileArray[i + 1]; // Creating a simple nameOfTheFile.root/graphic;1 to display on the tab
+
+             document.getElementById("TopBrowserId--aRootCanvas" + selecedTabID).innerHTML = ""; // Clearing the canvas
+             oTabElement.setAdditionalText(rootFileDisplayName); // Setting the tab file name
+             let finalJsonRoot = JSROOT.JSONR_unref(jsonAnswer.data); // Creating the graphic from the json
+             JSROOT.draw("TopBrowserId--aRootCanvas" + selecedTabID, finalJsonRoot, "colz"); // Drawing the graphic into the selected tab canvas
+
+             break;
```
Comment (bellenot) [inline ui5/browser/controller/Browser.controller.js]: Indentation

**Comment by linev** [comment]
Hi @Falcort,

As a **quick & dirty** solution it is ok, but we need much more advanced methods to work with ROOT structures. There are many global lists in ROOT, which should be accessed properly via RBrowser. In ROOT6 there is `TObject::Browse(TBrowser*)` method, which is implemented for many sub-classes. We need similar logic, but for arbitrary c++ classes - not only TObject.

One possible solution is implemented for THttpServer - it is so called TRootSniffer. It provides methods to scan many different classes without need to implement specialized `Browse()` method in the class itself. 

I hope, we can discuss this topic next week

**Comment by bellenot** [comment]
> As a **quick & dirty** solution it is ok, but we need much more advanced methods to work with ROOT structures. There are many global lists in ROOT, which should be accessed properly via RBrowser. In ROOT6 there is `TObject::Browse(TBrowser*)` method, which is implemented for many sub-classes. We need similar logic, but for arbitrary c++ classes - not only TObject.
> 
> One possible solution is implemented for THttpServer - it is so called TRootSniffer. It provides methods to scan many different classes without need to implement specialized `Browse()` method in the class itself.
> 
> I hope, we can discuss this topic next week

This is indeed more a proof of concept than a final solution. I propose to merge it as it is now and then review the whole concept (as several others, like directory navigation) when you'll be around next week. What do you think?

**Comment by linev** [comment]
Bertrand, you can merge PR.

**Pull request merged**